src/julia.h

```diff
@@ -179,6 +179,20 @@ STATIC_INLINE int jl_array_ndimwords(uint32_t ndims)
     return (ndims < 3 ? 0 : ndims-2);
 }
 
+typedef struct {
+    union {
```

Comment:
What code would be generated by the time

https://github.com/JuliaLang/julia/blob/ed760e40e972fc1c6bee81139c9dc919fc5a790f/src/init.c#L1300

happens? Still, it entirely true that just setting `pointerfree = false` might not work – but it did seem to work in a previous incarnation of this pull-request, without any comments to the contrary from anyone who looked at it. Which might simply have been due to no one looking at it.